### PR TITLE
stop using a singleton for SAML::SettingsService

### PR DESCRIPTION
**Pull request by aub**
We were using a singleton for storing the SAML settings, but then in the controllers we would get back the data from it and change it. Not great. This gets rid of the singleton and generates the data each time it's requested but still only queries the metadata endpoint once per app server instance.

### CODE REVIEW

**Comment by saneshark** [comment]
LGTM

File: lib/saml/settings_service.rb [hunk 1]
```diff
@@ -31,9 +34,9 @@ def settings
     end
 
     ## Makes an external web call to get IDP metadata and populates SETTINGS
-    def fetch_idp_metadata
+    def create_settings_from_metadata
       parser = OneLogin::RubySaml::IdpMetadataParser.new
-      parser.parse_remote(SAML_CONFIG['metadata_url'], true, settings: settings)
+      parser.parse(self.class.metadata, settings: settings)
```
Comment (omgitsbillryan) [inline lib/saml/settings_service.rb]: Skipping `parse_remote` in favor of `parse` means you skip cert validation.  From the ruby-saml source code:
```ruby
def parse_remote(url, validate_cert = true, options = {})
  idp_metadata = get_idp_metadata(url, validate_cert)
  parse(idp_metadata, options)
end
```
We _must_ validate the cert when fetching metadata.

Comment (aub) [inline lib/saml/settings_service.rb]: @omgitsbillryan, I just pushed a commit that makes it validate the SSL stuff when querying for the metadata. The Net::HTTP API is not the prettiest, but I believe this'll address your concern.

File: lib/saml/settings_service.rb [hunk 2]
```diff
@@ -3,14 +3,22 @@ module SAML
...
+    METADATA_URI = URI(SAML_CONFIG['metadata_url'])
+
     def initialize
-      @saml_settings ||= fetch_idp_metadata
+      @saml_settings = create_settings_from_metadata
+    end
+
+    def self.metadata
+      return @metadata if defined?(@metadata)
+      http = Net::HTTP.new(METADATA_URI.host, METADATA_URI.port)
```
Comment (omgitsbillryan) [inline lib/saml/settings_service.rb]: Wont `METADATA_URI.host` & `METADATA_URI.port` spit a `NoMethodError`?  `METADATA_URI` is just a string

Comment (aub) [inline lib/saml/settings_service.rb]: @omgitsbillryan It's not a string it's a URI.

**Pull request merged**